### PR TITLE
Fix parse test case on Python 3.12

### PR DESCRIPTION
**Pull request by JukkaL**
This fixes the test case testFStringWithFormatSpecifierExpression on Python 3.12.

Strip redundant empty string literal from AST generated from f-string. It's not generated on earlier Python versions and it seems fine to just drop it.

### CODE REVIEW

File: mypy/fastparse.py [hunk 1]
```diff
@@ -1535,6 +1535,12 @@ def visit_JoinedStr(self, n: ast3.JoinedStr) -> Expression:
         # Don't make unnecessary join call if there is only one str to join
         if len(strs_to_join.items) == 1:
             return self.set_line(strs_to_join.items[0], n)
+        elif len(strs_to_join.items) > 1:
+            last = strs_to_join.items[-1]
+            if isinstance(last, StrExpr) and last.value == "":
+                # 3.12 can add an empty literal at the end. Delete it for consistency
```
Comment (sobolevn) [inline mypy/fastparse.py:1541]: Interesting, why is that needed in CPython 🤔

Comment (AlexWaygood) [reply mypy/fastparse.py:1541]: > Interesting, why is that needed in CPython 🤔

It caused a crash over at flake8-bugbear too: https://github.com/PyCQA/flake8-bugbear/issues/393

Comment (hauntsaninja) [reply mypy/fastparse.py:1541]: If this has caused issues in two projects, should we report upstream?

Comment (AlexWaygood) [reply mypy/fastparse.py:1541]: Might be worth it, yeah

**Comment by AlexWaygood** [comment]
Looks like this might be changed in CPython, so maybe we should just wait for the next 3.12 beta rather than merging this:
- https://github.com/python/cpython/pull/106401

**Comment by JukkaL** [comment]
> maybe we should just wait for the next 3.12 beta rather than merging this

We can wait until the next beta, unless perhaps we want to enable 3.12 in CI before the next beta is available. I'll keep this open for now.

**Comment by AlexWaygood** [comment]
> unless perhaps we want to enable 3.12 in CI before the next beta is available.

For now we could maybe just skip the test. I can't remember the pytest equivalent off the top of my head, but for unittest I'd just decorate the test with:

```py
@skipIf(
    sys.version_info[:4] == (3, 12, 0, "beta") and sys.version_info[4] < 4,
    "Early 3.12 betas had a bug"
)
```

**Comment by JukkaL** [comment]
This is still failing for me on Python 3.12.0 beta 4. I'm leaning towards merging this. We can always revert this afterwards if it's no longer needed.

**Comment by AlexWaygood** [comment]
> This is still failing for me on Python 3.12.0 beta 4. I'm leaning towards merging this. We can always revert this afterwards if it's no longer needed.

Makes sense!
